OZ-069: Enum defined in user header not emitted in generated C

## Context

Exercising OZ-061 enum-in-header support. Moved `enum PXSensorType` from `.m` to `.h` so callers can use the constants across files. Discovered in `px-app`.

## Input

```objc
/* PXSensorBase.h */
enum PXSensorType {
        PXSensorTypeBase = 0,
        PXSensorTypeBarometer = 5
};

@interface PXSensorBase : OZObject
- (int)sensorType;
@end

/* PXBarometer.m — includes PXSensorBase.h via inheritance chain */
- (int)sensorType {
        return PXSensorTypeBarometer;
}
```

## Observed

Transpiler emits `return PXSensorTypeBarometer;` in `PXBarometer_ozm.c`, but does NOT emit the `enum PXSensorType { ... }` definition. GCC fails:
```
PXBarometer_ozm.c:54:16: error: 'PXSensorTypeBarometer' undeclared
```
Same issue for `PXDeviceState` and `PXLogLevel` when moved from `.m` to `.h`.

## Expected

When an enum is defined in a user header and its constants are used in a `.m` file, the transpiler should emit the enum definition in that file's generated `.c` output (or in the `_ozh.h` header).

## Workaround

Define the enum in each `.m` file that uses its constants (WA-007). Cross-file usage requires duplicating the enum definition.

## Metadata

| Field | Value |
|-------|-------|
| Filed by | DEV |
| Assigned to | rodrigopex |
| Blocking | NO |

--- a/tools/oz_transpile/emit.py
+++ b/tools/oz_transpile/emit.py
@@ -3068,6 +3068,18 @@ def _emit_patched_source(source_path: Path, module: OZModule,
     out.write(f'#include "{stem}_ozh.h"\n')
     for dep_stem in dep_stems:
         out.write(f'#include "{dep_stem}_ozh.h"\n')
+
+    # OZ-069: emit enum type_defs needed by the source but not in headers.
+    const_map = _type_def_constant_map(module)
+    header_td: list[str] = []
+    for c in module.classes.values():
+        for td in _header_type_defs_for_class(c, module):
+            if td not in header_td:
+                header_td.append(td)
+    src_type_defs = _scan_source_type_defs([rendered], const_map, header_td)
+    for td in src_type_defs:
+        out.write(f"\n{td}\n")
+
     for cls in classes:
         pc = pool_count_fn(cls.name) if pool_count_fn else 1
         if not isinstance(pc, int) or pc < 1:
@@ -3154,6 +3166,18 @@ def _emit_patched_orphan_source(orphan: OrphanSource, module: OZModule,
     out.write('#include "oz_dispatch.h"\n')
     for dep_stem in dep_stems:
         out.write(f'#include "{dep_stem}_ozh.h"\n')
+
+    # OZ-069: emit enum type_defs needed by orphan source.
+    const_map = _type_def_constant_map(module)
+    header_td: list[str] = []
+    for c in module.classes.values():
+        for td in _header_type_defs_for_class(c, module):
+            if td not in header_td:
+                header_td.append(td)
+    src_type_defs = _scan_source_type_defs([rendered], const_map, header_td)
+    for td in src_type_defs:
+        out.write(f"\n{td}\n")
+
     # Emit orphan statics not already preserved in the roundtrip
     for sv in orphan.statics:
         if sv.name not in orphan_static_names:

